src/libstd/net/addr.rs

```diff
@@ -632,7 +746,9 @@ pub trait ToSocketAddrs {
     ///
     /// # Errors
     ///
-    /// Any errors encountered during resolution will be returned as an `Err`.
+    /// Any errors encountered during resolution will be returned as an [`Err`].
```

Comment:
I looked around and it seems like the  `std::io::ErrorKind`s that might (commonly) come in the provided implementations are `InvalidInput`, `PermissionDenied`, and `NotFound`.
I will add this instance with this PR, but this is kind of an issue in general throughout the docs (#40322) and out of scope for what I wanted to achieve with this PR. Thanks for the pointer tho, this does sound rather silly!
(Sidenote: I will have to test this myself, but it seems like from [Windows' `decode_error_kind` implementation] that this might return an `io:Error` with `ErrorKind::Other` more often then it should/could on Windows??)

[Windows' `decode_error_kind` implementation]: https://github.com/rust-lang/rust/blob/450554ebf139d0b6ecde80f17c2ef89ff7de78bb/src/libstd/sys/windows/mod.rs#L69